### ISSUE TITLE
Unconditional usage of kFSEventStreamCreateFlagFileEvents breaks the build on < 10.7

### ISSUE THREAD

**Issue by barracuda156** [skaji/Mac-FSEvents#6]
All versions after 0.14 are broken on macOS 10.6 (and by implication on 10.5 as well).

```
--->  Building p5.28-mac-fsevents
Executing:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_perl_p5-mac-fsevents/p5.28-mac-fsevents/work/Mac-FSEvents-0.20" && /usr/bin/make -j6 -w all 
make: Entering directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_perl_p5-mac-fsevents/p5.28-mac-fsevents/work/Mac-FSEvents-0.20'
Running Mkbootstrap for FSEvents ()
"/opt/local/bin/perl5.28" "/opt/local/lib/perl5/5.28/ExtUtils/xsubpp"  -typemap '/opt/local/lib/perl5/5.28/ExtUtils/typemap' -typemap '/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_perl_p5-mac-fsevents/p5.28-mac-fsevents/work/Mac-FSEvents-0.20/typemap'  FSEvents.xs > FSEvents.xsc
chmod 644 "FSEvents.bs"
"/opt/local/bin/perl5.28" -MExtUtils::Command::MM -e 'cp_nonempty' -- FSEvents.bs blib/arch/auto/Mac/FSEvents/FSEvents.bs 644
cp lib/Mac/FSEvents/Event.pm blib/lib/Mac/FSEvents/Event.pm
cp lib/Mac/FSEvents.pm blib/lib/Mac/FSEvents.pm
Please specify prototyping behavior for FSEvents.xs (see perlxs manual)
mv FSEvents.xsc FSEvents.c
/usr/bin/gcc-4.2 -c   -fno-common -DPERL_DARWIN -pipe -Os -fno-strict-aliasing -fstack-protector -I/opt/local/include -DPERL_USE_SAFE_PUTENV -arch ppc -isysroot/ -O3   -DVERSION=\"0.20\" -DXS_VERSION=\"0.20\"  "-I/opt/local/lib/perl5/5.28/darwin-thread-multi-2level/CORE"   FSEvents.c
In file included from FSEvents.xs:242:
const-c.inc: In function ‘constant’:
const-c.inc:83: error: ‘kFSEventStreamCreateFlagFileEvents’ undeclared (first use in this function)
const-c.inc:83: error: (Each undeclared identifier is reported only once
const-c.inc:83: error: for each function it appears in.)
make: *** [FSEvents.o] Error 1
make: Leaving directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_perl_p5-mac-fsevents/p5.28-mac-fsevents/work/Mac-FSEvents-0.20'
Command failed:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_perl_p5-mac-fsevents/p5.28-mac-fsevents/work/Mac-FSEvents-0.20" && /usr/bin/make -j6 -w all 
Exit code: 2
Error: Failed to build p5.28-mac-fsevents: command execution failed
```

`kFSEventStreamCreateFlagFileEvents` was introduced in 10.7, but is used in the code unconditionally.

**Comment by barracuda156** [comment]
@skaji Could you please take a look? It will be nice to have it working again.

**Comment by skaji** [comment]
Thank you for your feedback.

While I understand the importance of ensuring compatibility across various platforms,
unfortunately, I am unable to support older versions of the OS moving forward.
One of the main reasons is that I no longer have the necessary test environments available to ensure the quality and stability of the module on these outdated OS.

I encourage you to consider updating to a more recent version of the OS.

**Comment by barracuda156** [comment]
@skaji While you do not have to, of course, it is perfectly possible to test 10.5/10.6 in a VM on any modern Intel and probably Apple Silicon Mac. You could even tests `ppc` builds via Rosetta.

P. S. PowerPC cannot run anything beyond 10.5 and pre-release 10.6, so updating OS is not a feasible choice.
